lib/trace_events.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const { hasTracing } = process.binding('config');
+const kHandle = Symbol('handle');
+const kEnabled = Symbol('enabled');
+const kCategories = Symbol('categories');
+
+const {
+  ERR_TRACE_EVENTS_UNAVAILABLE,
+  ERR_INVALID_ARG_TYPE
+} = require('internal/errors').codes;
+
+if (!hasTracing)
+  throw new ERR_TRACE_EVENTS_UNAVAILABLE();
+
+const { CategorySet, getEnabledCategories } = process.binding('trace_events');
+const { customInspectSymbol } = require('internal/util');
+const { format } = require('util');
+
+class Tracing {
+  constructor(categories) {
+    this[kHandle] = new CategorySet(categories);
+    this[kCategories] = categories;
+  }
+
+  enable() {
+    if (!this[kEnabled]) {
+      this[kEnabled] = true;
+      this[kHandle].enable();
+    }
+  }
+
+  disable() {
+    if (this[kEnabled]) {
+      this[kEnabled] = undefined;
+      this[kHandle].disable();
+    }
+  }
+
+  get enabled() {
+    return Boolean(this[kEnabled]);
```

Comment:
Why `Boolean(this[kEnabled])`,couldn't you just initialize `this[kEnabled] = false;` in the constructor?

Reply:
No particular reason, either works

Reply:
> No particular reason, either works

I would prefer the one that doesn't involve type conversion. I think V8 will also prefer the monomorphism.